SNIPS/snip-86.md

```diff
@@ -0,0 +1,278 @@
+                  "expirationTime",
+                  "notBefore"
+                ],
+                "errorMessage": "Name must be one of 'version', 'address', 'statement', 'uri', 'nonce', 'issuedAt', 'expirationTime', 'notBefore'"
+              },
+              "type": {
+                "type": "string",
+                "enum": ["string", "felt"],
+                "errorMessage": "Type must be either 'string' or 'felt'"
+              }
+            },
+            "required": ["name", "type"],
+            "additionalProperties": false,
+            "errorMessage": "Items must include name and type"
+          },
+          "minItems": 6,
+          "maxItems": 8,
+          "uniqueItems": true,
+          "errorMessage": "Message must contain min 6-8 unique items "
+        },
+        "StarknetDomain": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "enum": ["name", "chainId", "version", "revision"],
+                "errorMessage": "Name must be one of 'name', 'chainId', 'version', 'revision'"
+              },
+              "type": {
+                "type": "string",
+                "enum": ["felt", "string"],
+                "errorMessage": "Type must be 'felt' or 'string'"
+              }
+            },
+            "required": ["name", "type"],
+            "additionalProperties": false,
+            "errorMessage": "Items must include name and type"
+          },
+          "minItems": 4,
+          "maxItems": 4,
+          "uniqueItems": true,
+          "errorMessage": "StarknetDomain must contain exactly 4 unique items"
+        }
+      },
+      "required": ["Message", "StarknetDomain"],
+      "additionalProperties": false,
+      "errorMessage": "Types must include Message and StarknetDomain"
+    }
+  },
+  "required": ["domain", "message", "primaryType", "types"],
+  "additionalProperties": false,
+  "errorMessage": "Data must include domain, message, primaryType, types"
+}
+```
+
+## Implementation
+
+### Full Flow of Sign-In Process
+
+1. Challenge Preparation: A Starknet application (the challenger) prepares a sign-in request based on the specified schema. This request includes the domain and message objects.
+
+2. Signature Request: The user's wallet application displays the sign-in request, ensuring the user understands what they are signing.
+
+3. User Consent and Signature: If the user consents to the sign-in request, they sign the message using their Starknet account. This signature is then sent back to the challenger.
+
+4. Verification: The challenger uses an RPC call to a Starknet node to verify the signature against the user's account using the isValidSignature method. Due to account abstraction, the exact signature scheme doesn’t need to be known by the challenger; it only needs to ensure that the signature is valid for the given account.
```

Comment:
why isValidSignature? This breaks snake case convention on Starknet

Reply:
This is a typo, will change it to `is_valid_signature`